apis/v1alpha1/clusterobservability_types.go

```diff
@@ -0,0 +1,295 @@
+}
+
+// TLSConfig defines TLS configuration for the OTLP HTTP exporter.
+// This mirrors the OpenTelemetry Collector configtls settings.
+type TLSConfig struct {
+	// CAFile is the path to the CA certificate file for server verification.
+	// +optional
+	CAFile string `json:"ca_file,omitempty"`
+
+	// CertFile is the path to the client certificate file for mutual TLS.
+	// +optional
+	CertFile string `json:"cert_file,omitempty"`
+
+	// KeyFile is the path to the client private key file for mutual TLS.
+	// +optional
+	KeyFile string `json:"key_file,omitempty"`
+
+	// Insecure controls whether to use insecure transport. Default is false.
+	// +optional
+	Insecure bool `json:"insecure,omitempty"`
+
+	// ServerName for TLS handshake. If empty, uses the hostname from endpoint.
+	// +optional
+	ServerName string `json:"server_name,omitempty"`
+}
+
+// SendingQueueConfig defines configuration for the sending queue.
+type SendingQueueConfig struct {
+	// Enabled controls whether the queue is enabled. Default is true.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// NumConsumers is the number of consumers that dequeue batches. Default is 10.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	NumConsumers *int `json:"num_consumers,omitempty"`
+
+	// QueueSize is the maximum number of batches allowed in queue at a given time. Default is 1000.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	QueueSize *int `json:"queue_size,omitempty"`
+}
+
+// RetryConfig defines retry configuration for failed requests.
+type RetryConfig struct {
+	// Enabled controls whether retry is enabled. Default is true.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// InitialInterval is the initial retry interval (e.g., "5s"). Default is 5s.
+	// +optional
+	InitialInterval string `json:"initial_interval,omitempty"`
+
+	// RandomizationFactor is the randomization factor for retry intervals (e.g., "0.5"). Default is 0.5.
+	// +optional
+	RandomizationFactor string `json:"randomization_factor,omitempty"`
+
+	// Multiplier is the multiplier for retry intervals (e.g., "1.5"). Default is 1.5.
+	// +optional
+	Multiplier string `json:"multiplier,omitempty"`
+
+	// MaxInterval is the maximum retry interval (e.g., "30s"). Default is 30s.
+	// +optional
+	MaxInterval string `json:"max_interval,omitempty"`
+
+	// MaxElapsedTime is the maximum elapsed time for retries (e.g., "5m"). Default is 5m.
+	// +optional
+	MaxElapsedTime string `json:"max_elapsed_time,omitempty"`
+}
+
+// ClusterObservabilitySpec defines the desired state of ClusterObservability.
+// This follows a simplified design using a single OTLP HTTP exporter for all signals.
+type ClusterObservabilitySpec struct {
+	// Signals defines which observability signals to collect and export.
+	// Must contain at least one signal type from: logs, traces, metrics, profiles
+	// +required
+	// +kubebuilder:validation:MinItems=1
+	// +listType=set
+	Signals []ObservabilitySignal `json:"signals"`
```

Comment:
I like the simplicity of disabling/enabling by signal type. 

However I would prefer if the CR design was future proof and allowed us to add more detailed "nobs" in the future if we have to (once we collect feedback from the end users). 

Perhaps we can change the ObservabilitySignal from string to a struct.